pkg/nodediscovery/nodediscovery.go

```diff
@@ -218,17 +221,47 @@ func (n *NodeDiscovery) UpdateCiliumNodeResource() {
 
 	ciliumClient := k8s.CiliumClient()
 
-	performUpdate := true
-	nodeResource, err := ciliumClient.CiliumV2().CiliumNodes().Get(context.TODO(), nodeTypes.GetName(), metav1.GetOptions{})
-	if err != nil {
-		performUpdate = false
-		nodeResource = &ciliumv2.CiliumNode{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: nodeTypes.GetName(),
-			},
+	for retryCount := 0; retryCount < maxRetryCount; retryCount++ {
```

Comment:
Thank you for the PR. This was on my TODO list as well. I think we don't need to rely in retries. We could use watchers instead. Something like this: https://github.com/kubernetes/kubernetes/pull/88935/commits/e053fdd08ae00d1bff3dd42a2c034f4810f8a55d#diff-934760ea1a4e195e49bcd10625cf4e46R384

Does it makes sense?